src/main/java/org/assertj/core/api/AbstractMapAssert.java

```diff
@@ -1378,6 +1382,100 @@ public SELF withThreadDumpOnError() {
+   *
+   * assertThat(characters).extractingFromEntries(e -&gt; e.getValue().getName())
+   *                       .containsOnly(&quot;Omer&quot;,
+   *                                     &quot;Marge&quot;,
+   *                                     &quot;Lisa&quot;,
+   *                                     &quot;Bart&quot;);
+   *
+   * </code></pre>
+   *
+   **
+   * @param extractor the extractor function to extract a value from an entry of the Map under test.
+   * @return a new assertion object whose object under test is the list of values extracted
+   */
+  @CheckReturnValue
+  public AbstractListAssert<?, List<?>, Object, ObjectAssert<Object>> extractingFromEntries
+  (Function<? super Map.Entry<K, V>, Object> extractor) {
+
+    isNotNull();
+    List<Object> objects = actual.entrySet().stream().map(extractor::apply)
+                                 .collect(toList());
+    return newListAssertInstance(objects).as(info.description());
+  }
+
+
+  /**
+   *   Use the given {@link Function}s to extract values from the {@link Map}'s {@link Map.Entry}s.
+   *   The extracted values are stored in a new list composed of
+   *   {@link Tuple}s (a simple data structure containing the extracted values),
+   *   this new list becoming the object under test.
+   *
+   * <p>
+   *  This method works as {@link AbstractMapAssert#extractingFromEntries(java.util.function.Function)} except
+   *  that it is designed to extract multiple values from the {@link Map} entries.
+   *  That's why here the new object under test is a List of Object but a List of Tuples.
+   *  <p><p>
+   * The Tuple data corresponds to the extracted values from the Map's entries, for instance if you pass functions
+   * extracting the "id", "name" and "email" values then each Tuple data will be composed of an id, a name and an email
+   * extracted from the entry of the Map (the Tuple's data order is the same as the given functions order).
+   *
+   * Let's take a look at an example to make things clearer :
+   * <pre><code class='java'> // Build a Map that associates family roles and name of the Simpson familly
+   *
+   * Map&lt;String, CartoonCharacter&gt; characters = new HashMap&lt;&gt;();
+   * characters.put(&quot;dad&quot;, new CartoonCharacter(&quot;Omer&quot;));
+   * characters.put(&quot;mom&quot;, new CartoonCharacter(&quot;Marge&quot;));
+   * characters.put(&quot;girl&quot;, new CartoonCharacter(&quot;Lisa&quot;));
+   * characters.put(&quot;boy&quot;, new CartoonCharacter(&quot;Bart&quot;));
+   *
+   * assertThat(characters).extractingFromEntries(e -&gt; e.getKey(), e -&gt; e.getValue().getName())
+   *                       .containsOnly(tuple(&quot;dad&quot;, &quot;Omer&quot;),
+   *                                     tuple(&quot;mom&quot;, &quot;Marge&quot;),
+   *                                     tuple(&quot;girl&quot;, &quot;Lisa&quot;),
+   *                                     tuple(&quot;boy&quot;, &quot;Bart&quot;));
+   *
+   * </code></pre>
```

Comment:
more after `tuple(&quot;boy&quot;, &quot;Bart&quot;));`

Reply:
Not sure to understand.